classes/class-wc-rest-connect-shipping-label-controller.php

```diff
@@ -61,6 +61,8 @@ public function register_routes() {
 	public function update_items( $request ) {
 		$request_body = $request->get_body();
 		$settings = json_decode( $request_body, true, WOOCOMMERCE_CONNECT_MAX_JSON_DECODE_DEPTH );
+		$this->settings_store->update_origin_address( $settings[ 'origin' ] );
```

Comment:
Before:
* Merchant enters his address.
* Submit address for normalization.
* Server returns normalized address, and **that** address is stored in the settings.
* Merchant is a rebel and picks the un-normalized address anyway.
* Merchant purchases a label.

In that case, the stored address would be the normalized one, even though the merchant explicitly selected the other one (as Kelly told me, it happens that sometimes the address normalization services give you an address that's plain wrong). The next time the merchant opens the Labels UI modal, the `Origin` step would be collapsed and with a ✔️ so he won't realise that the address is "wrong".

It's better to only store the address when the merchant is actually purchasing a label for that address. Not 100% related with this issue but I noticed that odd behaviour while implementing it.